### DISCUSSION TITLE
谈谈 Mastodon、Fediverse 和 ActivityPub

### DISCUSSION THREAD

**Discussion by giscus** [zu1k/blog#65] [Comments]
# 谈谈 Mastodon、Fediverse 和 ActivityPub

最近 Mastodon 爆火，让我们来谈谈 Mastodon、Fediverse 和 ActivityPub 等相关内容

https://zu1k.com/posts/tutorials/p2p/fediverse/

**Comment by conge** [comment]
非常好奇这些私有实例，这些联邦成员如何实现续存呢？就是如何获得续存的资源，说到底，就是钱从哪儿来，联邦与实例的利益如何分配？
感觉上，理不顺这些问题，松散的联邦崩溃成封闭的社区，是早晚的事儿呀

**Reply by zu1k** [reply]
通过众筹而不是商业赞助，理论上一个良好的模式是任何想要参与社交的人都应该为自己占用的资源付出相应的金钱，这份钱不应该由其他商业团体提供(广告公司)

**Reply by zu1k** [reply]
想起一句话，When something online is free, you’re not the customer, you’re the product

**Comment by YYY2357** [comment]
国内审查制度和局域网操作，确实把社交软件发展的路子堵住了，去年的抒发森林app火了然后寄了。

**Comment by zu1k** [comment]
现在可以通过 [@zu1k@zu1k.com](https://social.zu1k.com/@zu1k) 关注我，这是用 https://github.com/superseriousbusiness/gotosocial  搭建的实例

**Reply by crisqyxw** [reply]
好巧😂我的实例也是GoToSocial

**Comment by zu1k** [comment]
推荐文章

https://tidbits.com/2023/01/27/mastodon-a-new-hope-for-social-networking/

**Comment by zu1k** [comment]
给大家反馈一下，我购买的1核1G的vps被服务商强制停了，理由是占用太多资源。因为peer了近500个其他服务器，在网络和磁盘IO上占用了太多共享资源，被封机器可以理解。实际上现有的公共服务商大多都是独立服务器，用户数量大的一般数据库/缓存/对象存储都是由独立的集群来提供

**Comment by maooson** [comment]
我上周刚搭了一个[mastodon节点](https://chat.aoq.me)，[@ottimo@chat.aoq.me](https://chat.aoq.me/@ottimo)，确实ruby的性能是硬伤。gotosocial也关注了，准备等稳定了再切。最近cloudflare也出了一个[Fediverse Server](https://github.com/cloudflare/wildebeest)

**Reply by zu1k** [reply]
cloudflare这个ts的应该不错，可惜跟他家产品捆绑太严重了，等后续看看有没有人给它解解绑

**Comment by livid** [comment]
Shower thoughts: so many new social platforms, Bluesky, Damus, Mastodon, Farcaster...what if we can have a new identity-centric model instead of platform-centric. Instead of deciding which platform we post to, we will only need to care about what will be posted from our id.

**Reply by zu1k** [reply]
Use PGP/GPG, send and receive anywhere text can be pasted, truly identity-centric, with guaranteed integrity, confidentiality and authenticity.

**Comment by linkerlin** [comment]
Go实现的mastodon是有的：https://github.com/superseriousbusiness/gotosocial 还在活跃的开发中。
Java实现的也有。

**Comment by zu1k** [comment]
AI自动生成的播客，挺有意思：https://blog-files.zu1k.com/AI%E5%8D%9A%E5%AE%A2-Mastodon.wav
